src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,177 @@
+        return status;
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::SetMaskValue(EndpointId endpoint, const BitMask<AlarmMap> & mask)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    status               = Attributes::Mask::Set(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: writing  mask %x", status);
+        return status;
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask.Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::SetStateValue(EndpointId endpoint, BitMask<AlarmMap> state)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    status               = Attributes::State::Set(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: writing  state %x", status);
+        return status;
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state.Raw());
+
+    // Generate Notify evenet.
+    BitMask<AlarmMap> becameActive   = state;
+    BitMask<AlarmMap> becameInActive = ~state.Raw();
+    // This field SHALL be a copy of the Mask attribute when this event was generated.
+    BitMask<AlarmMap> eMask;
+    BitMask<AlarmMap> newState = becameActive;
+
+    status = GetMaskValue(endpoint, &eMask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+    SendNotifyEvent(endpoint, becameActive, becameInActive, eMask, newState);
+
+    return status;
+}
+
+bool RefrigeratorAlarmServer::SendNotifyEvent(EndpointId endpointId, BitMask<AlarmMap> becameActive,
+                                              BitMask<AlarmMap> becameInactive, BitMask<AlarmMap> state, BitMask<AlarmMap> mask)
+{
+    Events::Notify::Type event{ .active = becameInactive, .inactive = becameInactive, .state = state, .mask = mask };
+    EventNumber eventNumber;
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "[Notify] Unable to send notify event: %s [endpointId=%d]", error.AsString(), endpointId);
+        return false;
+    }
+
+    return true;
+}
+
+void RefrigeratorAlarmServer::ModifyEnabledAlarmsCommand(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                         const Commands::ModifyEnabledAlarms::DecodableType & commandData)
+{
+    auto & mask         = commandData.mask;
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    EmberAfStatus status;
+    BitMask<AlarmMap> state;
+
+    status = SetMaskValue(endpoint, mask);
```

Comment:
Per spec, an application is supposed to be able to decide whether it can support disabling the alarm, for the generic alarms clusters.

But specifically for refrigerator alarm, remote-disabling the alarm is disallowed, no?  I don't see that being enforced here.  Again: we need _tests_.  This part is trivial to test in a YAML.

Reply:
> But specifically for refrigerator alarm, remote-disabling the alarm is disallowed, no?

Spec says you cannot disarm ( alarm Active --> Inactive) the alarm which is triggered unless there is some physical action. It doesn’t mention you cannot enable/disable alarms?

I mean remote modification in `State` ( Active/Inactive ) is not allowed. But spec doesn’t mentioned about the `Mask` ( enable/disable ).

Reply:
The spec is pretty explicit that the mask is fixed, and disabling is only possible at the device itself.  Although as there does appear to be confusion, I'll make that clearer.

Reply:
@jadhavrohit924 Yes, what @marktrayer said.  The Mask value for this cluster is a fixed constant that cannot be changed.